core/api/src/debug/add-payment-request-all-wallet-invoice.ts

```diff
@@ -0,0 +1,145 @@
+/**
+ * how to run:
+ *	. ./.envrc && yarn ts-node \
+ *		--files \
+ *			-r tsconfig-paths/register \
+ *			-r src/services/tracing.ts \
+ *		src/debug/add-payment-request-all-wallet-invoices.ts
+ */
+
+import { LndService } from "@/services/lnd"
+import { lndsConnect } from "@/services/lnd/auth"
+import { isUp } from "@/services/lnd/health"
+import { setupMongoConnection } from "@/services/mongodb"
+import {
+  SemanticAttributes,
+  addEventToCurrentSpan,
+  asyncRunInSpan,
+  wrapAsyncToRunInSpan,
+} from "@/services/tracing"
+import { WalletInvoice } from "@/services/mongoose/schema"
+
+const main = async () => {
+  return asyncRunInSpan(
+    "debug.addPaymentRequestAllWalletInvoices",
+    {
+      attributes: {
+        [SemanticAttributes.CODE_FUNCTION]: "addPaymentRequestAllWalletInvoices",
+        [SemanticAttributes.CODE_NAMESPACE]: "debug",
+      },
+    },
+    async () => {
+      const initialNumberOfLnInvoicesMissingPaymentRequest =
+        await WalletInvoice.countDocuments({
+          paymentRequest: { $exists: false },
+        })
+
+      console.log(
+        `Number of wallet invoices missing payment request: ${initialNumberOfLnInvoicesMissingPaymentRequest}`,
+      )
+
+      const lndService = LndService()
+      if (lndService instanceof Error) return lndService
+      const pubkeys = lndService.listActivePubkeys()
+      for (const pubkey of pubkeys) {
+        console.log(`Adding payment requests for pubkey: ${pubkey}`)
+        await addPaymentRequestAllWalletInvoicesForPubkey({
+          pubkey,
+        })
+        console.log(`Finished adding payment requests for pubkey: ${pubkey}`)
+      }
+
+      const finalNumberOfLnInvoicesMissingPaymentRequest =
+        await WalletInvoice.countDocuments({
+          paymentRequest: { $exists: false },
+        })
+      console.log(
+        `Number of wallet invoices missing payment request: ${finalNumberOfLnInvoicesMissingPaymentRequest}`,
+      )
+      console.log(
+        `Number of invoices that were modified: ${
+          initialNumberOfLnInvoicesMissingPaymentRequest -
+          finalNumberOfLnInvoicesMissingPaymentRequest
+        }`,
+      )
+    },
+  )
+}
+
+const addPaymentRequestAllWalletInvoicesForPubkey = wrapAsyncToRunInSpan({
+  fnName: "addPaymentRequestAllWalletInvoicesForPubkey",
+  namespace: "debug",
+  fn: async ({ pubkey }: { pubkey: Pubkey }) => {
+    const lndService = LndService()
+    if (lndService instanceof Error) return lndService
+
+    const invoices = await lndService.listInvoices({ pubkey })
```

Comment:
```suggestion
    const invoices = lndService.listInvoices({ pubkey })
```

I guess await is not necessary here?

Reply:
Good call